src/store/Favorite/FavoriteProvider.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useReducer } from 'react';
+import { FavoriteItem, FavoriteItemsContext } from '../types';
+import FavoriteContext from './favorite-context';
+
+interface FavoriteProviderProps {
+  children: React.ReactNode;
+}
+
+enum ActionType {
+  ADD = 'ADD',
+  REMOVE = 'REMOVE',
+}
+
+interface Action {
+  type: ActionType.REMOVE;
+  payload: Pick<FavoriteItem, 'id'>;
+}
+
+interface ActionAdd {
+  type: ActionType.ADD;
+  payload: FavoriteItem;
+}
+
+const favoriteReducer = (
+  state: FavoriteItem[],
+  action: Action | ActionAdd,
+): FavoriteItem[] => {
+  switch (action.type) {
+    case ActionType.ADD:
+      return [...state, action.payload];
+
+    case ActionType.REMOVE:
+      return state.filter(item => item.id !== action.payload.id);
+
+    default:
+      return state;
+  }
+};
+
+const FavoriteProvider: React.FC<FavoriteProviderProps> = props => {
+  const savedFavorites = sessionStorage.getItem('favoriteItems');
+  const initialState: FavoriteItem[] = savedFavorites
+    ? JSON.parse(savedFavorites)
+    : [];
+
+  const [favoriteState, dispatchFavoriteAction] = useReducer(
+    favoriteReducer,
+    initialState,
+  );
+
+  useEffect(() => {
+    sessionStorage.setItem('favoriteItems', JSON.stringify(favoriteState));
```

Comment:
It should be vice versa. Initially you should add items from **storage** to context **state**. Otherwise your storage will be cleared when reloading the page

Reply:
it's not cleared when reloading the page